peekingduck/pipeline/nodes/base.py

```diff
@@ -167,78 +126,36 @@ def check_valid_choice(
         if self.config[key] not in choices:
             raise ValueError(f"{key} must be one of {choices}")
 
-    def _check_above_value(
-        self, key: Union[str, List[str]], value: Number, inclusive: bool
-    ) -> None:
-        """Checks that configuration values specified by `key` is more than
-        (or equal to) the specified `value`.
-
-        Args:
-            key (Union[str, List[str]]): The specified key or list of keys.
-            value (Number): The specified value.
-            inclusive (bool): If `True`, compares `config[key] >= value`. If
-                `False`, compares `config[key] > value`.
-
-        Raises:
-            TypeError: `key` type is not in (List[str], str).
-            ValueError: If the configuration value is less than (or equal to)
-                `value`.
-        """
-        method = operator.ge if inclusive else operator.gt
-        extra_reason = " or equal to" if inclusive else ""
-        self._compare(key, value, method, reason=f"more than{extra_reason} {value}")
-
-    def _check_below_value(
-        self, key: Union[str, List[str]], value: Number, inclusive: bool
-    ) -> None:
-        """Checks that configuration values specified by `key` is more than
-        (or equal to) the specified `value`.
-
-        Args:
-            key (Union[str, List[str]]): The specified key or list of keys.
-            value (Number): The specified value.
-            inclusive (bool): If `True`, compares `config[key] <= value`. If
-                `False`, compares `config[key] < value`.
-
-        Raises:
-            TypeError: `key` type is not in (List[str], str).
-            ValueError: If the configuration value is less than (or equal to)
-                `value`.
-        """
-        method = operator.le if inclusive else operator.lt
-        extra_reason = " or equal to" if inclusive else ""
-        self._compare(key, value, method, reason=f"less than{extra_reason} {value}")
-
-    def _check_within_bounds(
+    def _check_within_bounds(  # pylint: disable=too-many-arguments
         self,
         key: Union[str, List[str]],
-        bounds: Tuple[Number, Number],
-        includes: Tuple[bool, bool],
+        lower: Number,
+        upper: Number,
+        lower_openness: str,
+        upper_openness: str,
```

Comment:
Initial thought process behind choosing `openness` as the interval can be "open" or "closed". Avoided "bracket" as it may be misunderstood to refer to only `[` in a pedantic way.

I was having trouble finding a word that can include both `(` and `[` or is "bracket" good enough?

Reply:
I understand `(` to be `round bracket` and `[` to be `square bracket`, although the former is aka `parenthesis`.
https://en.wikipedia.org/wiki/Bracket also mentions round/square brackets.

Just think `openness` is too mathematical and may confuse the general lay-programmer.

Reply:
Changed to use left and right bracket